src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,22 +46,71 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return blockNum.getBlockNum() + ", " + streetNum.getStreetNum() + ", " + 
+        		unitNum.getUnitNum() + ", " + postalCode.getPostalCode();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
```

Comment:
good that you update the equals check. you can also delegate the equality check to each component classes